src/pages/instances/forms/DuplicateInstanceForm.tsx

```diff
@@ -0,0 +1,205 @@
+          close();
+        })
+        .catch((e) => {
+          toastNotify.failure(
+            `Instance creation for "${values.instanceName}" failed.`,
+            e,
+          );
+          close();
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={close}
+      className="duplicate-instances-modal"
+      title="Duplicate Instance"
+      buttonRow={
+        <>
+          <Button
+            appearance="base"
+            className="u-no-margin--bottom"
+            type="button"
+            onClick={close}
+          >
+            Cancel
+          </Button>
+          <ActionButton
+            appearance="positive"
+            className="u-no-margin--bottom"
+            loading={formik.isSubmitting}
+            disabled={
+              !formik.isValid || storagePoolsLoading || clusterMembersLoading
+            }
+            onClick={() => void formik.submitForm()}
+          >
+            Duplicate
+          </ActionButton>
+        </>
+      }
+    >
+      <Form onSubmit={formik.handleSubmit}>
+        <Input
+          {...formik.getFieldProps("instanceName")}
+          type="text"
+          label="New instance name"
+        />
+        {isClustered && (
+          <Select
+            {...formik.getFieldProps("targetClusterMember")}
+            id="targetClusterMember"
+            label="Target cluster member"
+            options={clusterMembers.map((clusterMember) => {
+              return {
+                label: clusterMember.server_name,
+                value: clusterMember.server_name,
+              };
+            })}
+          />
+        )}
+        <Select
+          {...formik.getFieldProps("targetStoragePool")}
+          id="storagePool"
+          label="Storage pool"
+          options={storagePools.map((storagePool) => {
+            return {
+              label: storagePool.name,
+              value: storagePool.name,
+            };
+          })}
+        />
+        <Input
+          {...formik.getFieldProps("allowInconsistent")}
+          type="checkbox"
+          label="Ignore copy errors for volatile files"
+          error={
+            formik.touched.allowInconsistent
+              ? formik.errors.allowInconsistent
+              : null
+          }
+        />
+        <Input
+          {...formik.getFieldProps("instanceOnly")}
+          type="checkbox"
+          label="Copy without instance snapshots"
+          error={
+            formik.touched.instanceOnly ? formik.errors.instanceOnly : null
+          }
+        />
+        <Input
+          {...formik.getFieldProps("stateful")}
+          type="checkbox"
+          label="Copy stateful"
+          error={formik.touched.stateful ? formik.errors.stateful : null}
+        />
+        {/* hidden submit to enable enter key in inputs */}
+        <Input type="submit" hidden value="Hidden input" />
+      </Form>
+    </Modal>
+  );
+};
+
+export default DuplicateInstanceForm;
```

Comment:
```suggestion
import { FC, useState } from "react";
import { LxdInstance } from "types/instance";
import { useFormik } from "formik";
import { useToastNotification } from "context/toastNotificationProvider";
import {
  ActionButton,
  Button,
  Form,
  Input,
  Modal,
  Select,
} from "@canonical/react-components";
import * as Yup from "yup";
import { createInstance } from "api/instances";
import { isClusteredServer } from "util/settings";
import { useSettings } from "context/useSettings";
import { useQuery } from "@tanstack/react-query";
import { queryKeys } from "util/queryKeys";
import { fetchStoragePools } from "api/storage-pools";
import { fetchClusterMembers } from "api/cluster";
import { Link } from "react-router-dom";
import { instanceNameValidation } from "util/instances";
import { LxdDiskDevice } from "types/device";
import InstanceLink from "pages/instances/InstanceLink";
import { useEventQueue } from "context/eventQueue";

interface Props {
  instance: LxdInstance;
  close: () => void;
}

export interface LxdInstanceDuplicate {
  instanceName: string;
  targetClusterMember?: string;
  targetStoragePool?: string;
  allowInconsistent?: boolean;
  instanceOnly?: boolean;
  stateful?: boolean;
}

const DuplicateInstanceForm: FC<Props> = ({ instance, close }) => {
  const toastNotify = useToastNotification();
  const { data: settings } = useSettings();
  const isClustered = isClusteredServer(settings);
  const controllerState = useState<AbortController | null>(null);
  const eventQueue = useEventQueue();

  const { data: storagePools = [], isLoading: storagePoolsLoading } = useQuery({
    queryKey: [queryKeys.storage],
    queryFn: () => fetchStoragePools(instance.project),
  });

  const { data: clusterMembers = [], isLoading: clusterMembersLoading } =
    useQuery({
      queryKey: [queryKeys.cluster],
      queryFn: fetchClusterMembers,
      enabled: isClustered,
    });

  const notifySuccess = (instanceName: string) => {
    const newInstance = (
      <Link to={`/ui/project/${instance.project}/instance/${instanceName}`}>
        {instanceName}
      </Link>
    );
    toastNotify.success(
      <>
        Created the instance <b>{newInstance}</b>.
      </>,
    );
  };

  const formik = useFormik<LxdInstanceDuplicate>({
    initialValues: {
      instanceName: instance.name + "-duplicate",
      allowInconsistent: false,
      instanceOnly: false,
      stateful: false,
      targetClusterMember: clusterMembers[0]?.server_name,
      targetStoragePool:
        (instance.devices["root"] as LxdDiskDevice)?.pool ??
        storagePools[0]?.name,
    },
    validationSchema: Yup.object().shape({
      instanceName: instanceNameValidation(
        instance.project,
        controllerState,
        instance.name,
      ).required(),
    }),
    onSubmit: (values) => {
      createInstance(
        JSON.stringify({
          description: instance.description,
          name: values.instanceName,
          source: {
            allow_inconsistent: values.allowInconsistent,
            instance_only: values.instanceOnly,
            source: instance.name,
            type: "copy",
          },
          stateful: values.stateful,
          devices: {
            ...instance.devices,
            root: {
              path: "/",
              type: "disk",
              pool: values.targetStoragePool,
            },
          },
        }),
        instance.project,
        values.targetClusterMember,
      )
        .then((operation) => {
          eventQueue.set(
            operation.metadata.id,
            () => {
              notifySuccess(values.instanceName);
            },
            (msg) =>
              toastNotify.failure(
                "Instance duplication failed",
                new Error(msg),
                <InstanceLink instance={instance} />,
              ),
            () => close(),
          );
        })
        .catch((e) => {
          toastNotify.failure(
            `Instance creation for "${values.instanceName}" failed.`,
            e,
          );
          close();
        });
    },
  });

  return (
    <Modal
      close={close}
      className="duplicate-instances-modal"
      title="Duplicate Instance"
      buttonRow={
        <>
          <Button
            appearance="base"
            className="u-no-margin--bottom"
            type="button"
            onClick={close}
          >
            Cancel
          </Button>
          <ActionButton
            appearance="positive"
            className="u-no-margin--bottom"
            loading={formik.isSubmitting}
            disabled={
              !formik.isValid || storagePoolsLoading || clusterMembersLoading
            }
            onClick={() => void formik.submitForm()}
          >
            Duplicate
          </ActionButton>
        </>
      }
    >
      <Form onSubmit={formik.handleSubmit}>
        <Input
          {...formik.getFieldProps("instanceName")}
          type="text"
          label="New instance name"
          error={
            formik.touched.instanceName ? formik.errors.instanceName : null
          }
        />
        {isClustered && (
          <Select
            {...formik.getFieldProps("targetClusterMember")}
            id="targetClusterMember"
            label="Target cluster member"
            options={clusterMembers.map((clusterMember) => {
              return {
                label: clusterMember.server_name,
                value: clusterMember.server_name,
              };
            })}
          />
        )}
        <Select
          {...formik.getFieldProps("targetStoragePool")}
          id="storagePool"
          label="Storage pool"
          options={storagePools.map((storagePool) => {
            return {
              label: storagePool.name,
              value: storagePool.name,
            };
          })}
        />
        <Input
          {...formik.getFieldProps("allowInconsistent")}
          type="checkbox"
          label="Ignore copy errors for volatile files"
          error={
            formik.touched.allowInconsistent
              ? formik.errors.allowInconsistent
              : null
          }
        />
        <Input
          {...formik.getFieldProps("instanceOnly")}
          type="checkbox"
          label="Copy without instance snapshots"
          error={
            formik.touched.instanceOnly ? formik.errors.instanceOnly : null
          }
        />
        <Input
          {...formik.getFieldProps("stateful")}
          type="checkbox"
          label="Copy stateful"
          error={formik.touched.stateful ? formik.errors.stateful : null}
        />
        {/* hidden submit to enable enter key in inputs */}
        <Input type="submit" hidden value="Hidden input" />
      </Form>
    </Modal>
  );
};

export default DuplicateInstanceForm;
```
Adding error handling and support for the disk device

Reply:
Oof, thank you for this.

Reply:
I get this error when trying to duplicate an instance for the first time.
It is interesting because if I attempt to duplicate after this, I can. Or if I manually set the default, I can. This also resets every time the page is refreshed.
![image](https://github.com/user-attachments/assets/9decd0d6-c34f-44ad-98e3-d5a2505acd68)

Reply:
I updated the suggestion above. There was missing the `enableReinitialize` setting on formik and the notification should be with an info and closing the modal right after submit and only once the background task to duplicate succeeds or fails we should show the final notification.